### ISSUE TITLE
the metric for Agent of Stackdriver doesn't return 'instance name' correctly

### ISSUE THREAD

**Issue by jwlee98** [grafana/grafana#18750]
**What happened**:
When I tested Grafana with Stackdriver, it worked well usually. But, the metric for Agent doesn't return 'instance name' like below (Explorer in Grafana -> Service : Agent, then choose any metric.) - It means we could not draw graph for agent based on 'instance name'(only agent has memory data)

```
{"results":{"B":{"refId":"B","meta":{"alignmentPeriod":60,"groupBys":[],"metricLabels":{},"rawQuery":"aggregation.alignmentPeriod=%2B60s\u0026aggregation.crossSeriesReducer=REDUCE_NONE\u0026aggregation.perSeriesAligner=ALIGN_INTERPOLATE\u0026filter=metric.type%3D%22agent.googleapis.com%2Fcpu%2Fload_1m%22\u0026interval.endTime=2019-08-27T14%3A28%3A05Z\u0026interval.startTime=2019-08-27T08%3A28%3A05Z\u0026view=FULL","resourceLabels":{"instance_id":["4115361936374834423","536450093317781497","1758319408006985720","8185767047694546937","8646741266157584654","5991586440756240602","4664485625565510539","6996926619894031488"],"project_id":["jwlee-dialogflow-test01","jwlee-myproject-01"],"zone":["us-central1-c","asia-northeast1-a","asia-northeast1-b","asia-northeast1-c","asia-northeast2-b","asia-northeast2-c","us-central1-f"]},"resourceTypes":["gce_instance"]},"series":[{"name":"agent.googleapis.com/cpu/load_1m","points":[[0,1566894605000],[0.11525952910151341,1566894665000]
```

It's sample for compare(it's data for 'compute', not 'agent)
```
{"results":{"B":{"refId":"B","meta":{"alignmentPeriod":60,"groupBys":[],"metricLabels":{"instance_name":["aef-default-20180824t194433-n2w2","vm-group-1-7kfx","vm-group-1-3t3l","vm-group-1-hf2v","shell-instance-1","mysql-test-2","f1-test-instance-1","bindplane-collector","win-2019-instance-1","g1-nginx-instance-2","mysql-blndplane-01","instance-2","gcesysprep-win2016","gke-kub0standard-cluster-default-pool-0f42aa63-kknf","gke-kub0standard-cluster-default-pool-0f42aa63-w7zc","gke-kub0standard-cluster-default-pool-0f42aa63-1xx9","iap-instance-2","grafana-1-vm"]}
```

**What you expected to happen**:
When I choose the metic for agent, I want to catch 'instance name' as well. It means we could not draw graph for agent based on 'instance name'(only agent has memory data)

**How to reproduce it (as minimally and precisely as possible)**:
Explorer in Grafana -> Service : Agent, then choose any metric.  

**Anything else we need to know?**:

**Environment**:
- Grafana version: Grafana v6.1.6 (commit: cf9cb45)
- Data source type & version: Stackdriver
- OS Grafana is installed on: Debian (9.8)
- User OS & Browser: Mac & Chrome
- Grafana plugins: Stackdriver plugin 1.0.0
- Others:

**Comment by igorpeshansky** [comment]
For anyone who'll be working on the implementation.
The Stackdriver Monitoring API supports grouping by metadata labels (`system_labels`, which include instance names, and `user_labels`, which are basically instance tags). When you add a groupby `metadata.system_labels.name` to any query against a `gce_instance` resource, the response should contain a `metadata` section, e.g.:
```
"metadata":{
  "system_labels":{
    "name":"my-instance"
  }
}
```
The Grafana Stackdriver source should support extracting the `metadata` part of the response and manifesting it as selectable labels in the UI.

**Comment by sunker** [comment]
From what I can tell by reading the [api documentation](https://cloud.google.com/monitoring/api/metrics_agent#agent-agent), there are no such label as `instance_name` being supported for `agent` right now.

**Comment by igorpeshansky** [comment]
The agent does not (and will not) ingest that label. The label is added through other means and manifests in the metadata section of the response, as I mentioned in https://github.com/grafana/grafana/issues/18750#issuecomment-525956833.

**Comment by sunker** [comment]
I see what you mean now @igorpeshansky. Just opened a new feature request, #18887. Will look into this sometime soon.

**Comment by sunker** [comment]
fixed in #18887
